test/16-fairness-tax.ts

```diff
@@ -0,0 +1,346 @@
+            bonusTimestamp: `${lastDepositTimestampForTaxInit} --> ${lastDepositTimestampForTaxFinal}`
+        })
+    })
+
+    it(`TAX TIMESTAMP: The fairness tax should reset for a token when a use deposits > 5% of their current staked amount`, async function() {
+        const { user1 } = await getNamedSigners(hre)
+        const summitToken = await getSummitToken()
+
+        const lastDepositTimestampForTaxInit = await cartographerGet.tokenLastDepositTimestampForTax(user1.address, summitToken.address)
+
+        const _userStakedAmount = await cartographerGet.userTokenStakedAmount(user1.address, summitToken.address)
+        const _taxResetOnDepositBP = await cartographerGet.taxResetOnDepositBP()
+        const _newDepositAmount = (_userStakedAmount * _taxResetOnDepositBP / 10000) + 1
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: summitToken.address,
+            elevation: OASIS,
+            amount: e18(_newDepositAmount),
+        })
+        const depositTimestamp = await getTimestamp()
+        const lastDepositTimestampForTaxFinal = await cartographerGet.tokenLastDepositTimestampForTax(user1.address, summitToken.address)
+
+        expect(lastDepositTimestampForTaxInit).to.not.equal(lastDepositTimestampForTaxFinal)
+        expect(lastDepositTimestampForTaxFinal).to.equal(depositTimestamp)
+
+        console.log({
+            bonusTimestamp: `${lastDepositTimestampForTaxInit} --> ${lastDepositTimestampForTaxFinal}`
+        })
+    })
+
+    it(`STAKED TOTAL AMOUNT FOR TOKEN: The user's total staked amount for token should be calculated correctly across all elevations`, async function() {
+        const { user1 } = await getNamedSigners(hre)
+        const summitToken = await getSummitToken()
+        const oasisDepositAmount = 5;
+        const plainsDepositAmount = 10;
+
+        const userTotalStakedAmountBefore = await cartographerGet.userTokenStakedAmount(user1.address, summitToken.address)
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: summitToken.address,
+            elevation: OASIS,
+            amount: e18(oasisDepositAmount),
+        })
+
+        await rolloverRound(PLAINS)
+        await cartographerMethod.switchTotem({
+            user: user1,
+            elevation: PLAINS,
+            totem: 0,
+        })
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: summitToken.address,
+            elevation: PLAINS,
+            amount: e18(plainsDepositAmount),
+        })
+
+        const userTotalStakedAmountAfter = await cartographerGet.userTokenStakedAmount(user1.address, summitToken.address)
+        expect(userTotalStakedAmountAfter).to.equal(userTotalStakedAmountBefore + oasisDepositAmount + plainsDepositAmount)
+    })
+
+    it(`TAX BP: The fairness tax decreases correctly to the 0% for native farms, over the duration of the tax decay`, async function() {
```

Comment:
I think this test is really testing `TAX BP: The minimum fairness tax of native tokens should be 0%` since it doesn't actually test the tax decreatisg